include/alpaka/block/shared/st/BlockSharedMemStMember.hpp

```diff
@@ -122,18 +42,25 @@ namespace alpaka
             //-----------------------------------------------------------------------------
             static auto declareVar(BlockSharedMemStMember<TDataAlignBytes> const& smem) -> T&
             {
-                smem.template alloc<T>();
-                return smem.template getLatestVar<T>();
+                auto* data = smem.template getVar<T>(TuniqueId);
+
+                if(!data)
+                {
+                    smem.template alloc<T>(TuniqueId);
+                    data = smem.template getVar<T>(TuniqueId);
```

Comment:
`BlockSharedMemStMember` can only be used in accelerators which only run one thread per block, thus `getLatestVar` would be sufficient here, no need to iterate through all variables again.

Reply:
This is required to fulfill the requirement that some allocations of shared memory (same template signature) will not allocate with each call new memory.

Reply:
> BlockSharedMemStMember can only be used in accelerators which only run one thread per block, thus getLatestVar would be sufficient here, no need to iterate through all variables again.

Get the last variable is not allowed to be called from outside anymore. The reason is that the last element in the memory blob is always the message header. 
IMO the performance drawback is very tiny.

Reply:
This what the first `getVar` is for. The second one happens after a new allocation was already found to be required.

Reply:
The message header is aligned and the last element in the memory blob, therefore I do not know the starting address of the last allocated variable.